cmd/kubeadm/app/apis/kubeadm/validation/validation.go

```diff
@@ -407,3 +408,25 @@ func ValidateIgnorePreflightErrors(ignorePreflightErrors []string, skipPreflight
 
 	return ignoreErrors, allErrs.ToAggregate()
 }
+
+// ValidateSocketPath validates format of socket path or url
+func ValidateSocketPath(socket string, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	u, err := url.Parse(socket)
+	if err != nil {
+		return append(allErrs, field.Invalid(fldPath, socket, fmt.Sprintf("socket parsing error: %+v", err)))
+	}
+
+	if u.Scheme != "" && u.Scheme != kubeadmapiv1alpha3.DefaultUrlScheme {
+		return append(allErrs, field.Invalid(fldPath, socket, fmt.Sprintf("url scheme %s is not supported", u.Scheme)))
+	}
+
+	if u.Scheme == "" || u.Scheme == "unix" {
+		if !filepath.IsAbs(u.Path) {
```

Comment:
if `Scheme == ""` then `Path` **should be** populated and you can `IsAbs` test directly:
```
if u.Scheme == "" && !filepath.IsAbs(u.Path) {
	return append(allErrs, field.Invalid(fldPath, socket, fmt.Sprintf("path is not absolute: %q", u.Path))
	// ^ please add path output
}
```

i could be missing something but i don't think we should do `u.Scheme == "unix"` on Windows.